File: app/src/main/java/com/snap/model/shading/UniformBinding.java
package com.snap.model.shading;

import android.opengl.GLES20;

public class UniformBinding {

    private final String uniformName;
    private final int uniformHandle;

    public UniformBinding(int programHandle, String uniformName) {
        this.uniformName = uniformName;
        this.uniformHandle = GLES20.glGetUniformLocation(programHandle, uniformName);
    }

    public String getUniformName() {
        return uniformName;
    }

    public void bindUniformMatrix4fv(final float[] matrix) {
        GLES20.glUniformMatrix4fv(uniformHandle, 1, false, matrix, 0);
    }

    public void bindUniform3f(final float first, final float second, final float third) {
        GLES20.glUniform3f(uniformHandle, first, second, third);
    }

    public void bindUniform4f(final float first, final float second, final float third, final float fourth) {
        GLES20.glUniform4f(uniformHandle, first, second, third, fourth);
    }
}

File: app/src/main/java/com/snap/model/shading/UniformBinding.kt
package com.snap.model.shading

import android.opengl.GLES20

class UniformBinding(programHandle: Int, uniformName: String) {

    private val uniformHandle: Int = GLES20.glGetUniformLocation(programHandle, uniformName)

    fun bindUniformMatrix4fv(matrix: FloatArray) {
        GLES20.glUniformMatrix4fv(uniformHandle, 1, false, matrix, 0)
    }

    fun bindUniform3f(first: Float, second: Float, third: Float) {
        GLES20.glUniform3f(uniformHandle, first, second, third)
    }

    fun bindUniform4f(first: Float, second: Float, third: Float, fourth: Float) {
        GLES20.glUniform4f(uniformHandle, first, second, third, fourth)
    }
}
